src/main/java/me/coley/recaf/util/Classpath.java

```diff
@@ -138,38 +146,112 @@ public void scan(ClassLoader classLoader) {
+						Set modules = new HashSet();
+
+						Object ourModule = getModule.invoke(Classpath.class);
+						Object ourLayer = getLayer.invoke(ourModule);
+						if (ourLayer != null) {
+							modules.addAll((Set) layerModules.invoke(ourLayer));
+						}
+						modules.addAll(
+								(Set) layerModules
+										.invoke(lookup.findStatic(layer, "boot", MethodType.methodType(layer))
+												.invoke()));
+						for (ClassLoader c = Classpath.class.getClassLoader(); c != null; c = c.getParent()) {
+							modules.add(unnamedModule.invoke(c));
+						}
+
+						for (Object impl : modules) {
+							for (String name : (Set<String>) getPackages.invoke(impl)) {
+								export.invoke(impl, name);
+							}
+						}
+
+						Class<?> reflectionClass = Class.forName("jdk.internal.reflect.Reflection");
+						MethodHandle getMapHandle = lookup.findStaticGetter(reflectionClass, "fieldFilterMap", Map.class);
+						// Map is immutable, thanks Oracle
+						Map<Class<?>, Set<String>> fieldFilterMap = new HashMap<>((Map<Class<?>, Set<String>>) getMapHandle.invokeExact());
+						fieldFilterMap.remove(Field.class);
+						fieldFilterMap.remove(ClassLoader.class);
+						lookup.findStaticSetter(reflectionClass, "fieldFilterMap", Map.class)
+								.invokeExact(fieldFilterMap);
+
+						Method method = Class.forName("jdk.internal.loader.ClassLoaders").getDeclaredMethod("bootLoader");
+						method.setAccessible(true);
+						Object bootLoader = method.invoke(null);
+						Field field = bootLoader.getClass().getSuperclass().getDeclaredField("ucp");
+						field.setAccessible(true);
+
+						Object bootstrapClasspath = field.get(bootLoader);
+						if (bootstrapClasspath != null)
+							scanBootstrapClasspath(URLClassLoader.class.getDeclaredField("ucp"), classLoader, bootstrapClasspath);
+
+						// Now, scan all modules
+						Class<?> loadedModuleClass = Class.forName("jdk.internal.loader.BuiltinClassLoader$LoadedModule");
+						Method mref = loadedModuleClass.getDeclaredMethod("mref");
+						mref.setAccessible(true);
+						Class<?> mrefClass = mref.getReturnType();
+						Method openReader = mrefClass.getDeclaredMethod("open");
+						openReader.setAccessible(true);
+						Method closeReader = openReader.getReturnType().getDeclaredMethod("close");
+						closeReader.setAccessible(true);
+						Method listReader = openReader.getReturnType().getDeclaredMethod("list");
+						listReader.setAccessible(true);
+						Field bootModulesField = bootLoader.getClass().getSuperclass().getDeclaredField("packageToModule");
+						bootModulesField.setAccessible(true);
+						Collection<?> packageToModule = ((Map<String, ?>) bootModulesField.get(null)).values();
+						for (Object module : packageToModule) {
+							// Scan it!
+							Object ref = mref.invoke(module);
+							Object reader = openReader.invoke(ref);
+							Stream<String> list = (Stream<String>) listReader.invoke(reader);
+							list.parallel().filter(s -> s.endsWith(".class"))
```

Comment:
```suggestion
							list.filter(s -> s.endsWith(".class"))
```

[Should I always use a parallel stream when possible? - Stack Overflow](https://stackoverflow.com/a/20375622/6238618)